From 1d775562d16804139f43022b3eaa72e93851d431 Mon Sep 17 00:00:00 2001
From: Kevin McCarthy <kevin@8t8.us>
Date: Tue, 14 Jun 2016 13:11:56 -0700
Subject: [PATCH 45/45] Reset buffy->new for the current mailbox in IMAP.

1f840760e6e0 moved the buffy->new reset inside the STATUS processor.
Since the current mailbox is not STATUS'ed, it needs to be reset in
imap_buffy_check().

Thanks to Aaron Schrab for reporting this issue (and for helping test
tip).
---
 imap/imap.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/imap/imap.c b/imap/imap.c
index 1a51a2e..b8d9b0a 100644
--- a/imap/imap.c
+++ b/imap/imap.c
@@ -1511,14 +1511,20 @@ int imap_buffy_check (int force)
     mailbox->new = 0;
 
     if (imap_get_mailbox (mailbox->path, &idata, name, sizeof (name)) < 0)
+    {
+      mailbox->new = 0;
       continue;
+    }
 
     /* Don't issue STATUS on the selected mailbox, it will be NOOPed or
      * IDLEd elsewhere.
      * idata->mailbox may be NULL for connections other than the current
      * mailbox's, and shouldn't expand to INBOX in that case. #3216. */
     if (idata->mailbox && !imap_mxcmp (name, idata->mailbox))
+    {
+      mailbox->new = 0;
       continue;
+    }
 
     if (!mutt_bit_isset (idata->capabilities, IMAP4REV1) &&
         !mutt_bit_isset (idata->capabilities, STATUS))
-- 
2.8.2

